### ISSUE TITLE
Fully inline (.js + json manifest) service worker

### ISSUE THREAD

**Issue by hatgit** [w3c/ServiceWorker#1392]
I have an app that runs on a standalone basis from within a .html file as all the code and dependencies are inline including the javascript and css. However, even though the app works the same whether one is connected to the internet or not, it fails to pass the Progressive Web app audits that Lighthouse runs in Chrome Dev Tools, because a service work is not being registered. 

**Progressive Web App Dilemma**

I am trying to determine whether it would be possible to add a service worker that registers WITHOUT the need of having any additional file by bringing both the service worker related-javascript and json manifest code inline into the main .html file. Does anyone know if this would be possible? 

The reason is so that even when the page is saved offline one could install it locally and the browser would offer a user to save it as it would recognize it as a progress web app.

**Comment by mattto** [comment]
I don't think this is possible today because the SW needs to be its own URL. There are some proposed technologies like https://wicg.github.io/webpackage/draft-yasskin-wpack-bundled-exchanges.html which might someday meet your use case of saving offline and install locally.

**Comment by yoshisatoyanagisawa** [comment]
Upon https://developer.chrome.com/docs/web-platform/web-bundles, Web Bundles has been deprecated in Chrome 2023.

**Comment by monica-ch** [comment]
[Bug triage] Closing the issue as not planned, feel free to re-open if still needed.
